h/static/scripts/controllers/disable-on-submit-controller.js

```diff
@@ -0,0 +1,23 @@
+'use strict';
+
+const Controller = require('../base/controller');
+
+/**
+ * For this form, disable all submit elements (inputs or buttons) after
+ * the form has been submitted—helps prevent duplicate submission of form data.
+ */
+class DisableOnSubmit extends Controller {
+  constructor(element) {
+    super(element);
+
+    const submitEls = element.querySelectorAll('[type="submit"]');
```

Comment:
If a form has a `<button>` with no "type" attribute, the button will function as a submit button, but adding this controller to the form won't work as expected. I was thinking we could resolve that by changing the selector here to `[type="submit"],button:not([type])`. Thoughts?

Reply:
I could add this to be all-inclusive, _but_ here's the thing: I'm not sure I'm down with the idea of having a submit button in markup not in some way semantically identified as such. We have the control over our forms (we'd have to actively _add_ this to a form; it's opt-in)...can you give me an example where we'd _need_ to have a non-typed `button` as a form's submit-er? It seems like we have the power to A) use `input type="submit"` or B) if we must, `button type="submit"`. But perhaps you know of cases where this is not possible for some reason?

Reply:
I can't think of any cases where a form would have a submit button in h that cannot be explicitly marked as such. My concern was that someone might add one, without being aware of the preference to explicitly specify a type, then add this controller and not realize that it wasn't working.

I don't feel that it is that important, but just a hazard that someone could trip over in future. I would be OK if you decided not to handle this for now (leave the PR as-is), or to explicitly throw an error, or make the change suggested above.